### PR TITLE
[SPARK-42692][CONNECT] Implement `Dataset.toJSON`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr aims to implement Dataset.toJSON.

### Why are the changes needed?
Add Spark connect jvm client api coverage.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
- Add new test
- Manually checked Scala 2.13

### CODE REVIEW

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 1]
```diff
@@ -2777,7 +2778,11 @@ class Dataset[T] private[sql] (
   }
 
   def toJSON: Dataset[String] = {
-    throw new UnsupportedOperationException("toJSON is not implemented.")
+    sparkSession.newDataset(StringEncoder) { builder =>
```
Comment (hvanhovell) [inline connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala:2781]: Why is this better over `project(to_json(col(*))).as(StringEncoder)`?

File: connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/ClientE2ETestSuite.scala [hunk 2]
```diff
@@ -631,6 +631,19 @@ class ClientE2ETestSuite extends RemoteSparkSession with SQLHelper {
     val otherPlan = spark.sql("select 1")
     assert(plan.sameSemantics(otherPlan))
   }
+
+  test("toJSON") {
+    val expected = Array(
+      "{\"b\":0.0,\"id\":0,\"d\":\"world\",\"a\":0}",
```
Comment (hvanhovell) [inline connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/ClientE2ETestSuite.scala:637]: Use raw strings here. That is easier to read.
```scala
val expected = Array(
  """{"b": 0.0,"id":0,"d":"world","a":0}""",
  ...
)
```

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala (hunk 1 above)
Comment (LuciferYang) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala:2781]: Sorry ... what is project?

Comment (LuciferYang) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala:2781]: Thanks . Sorry I didn't understand what  `project` is before ...

Comment (LuciferYang) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala]: fixed

File: connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/ClientE2ETestSuite.scala (hunk 2 above)
Comment (LuciferYang) [reply connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/ClientE2ETestSuite.scala]: done

**Review by hvanhovell — APPROVED** [review]
LGTM

**Comment by amaliujia** [comment]
LGTM

**Comment by LuciferYang** [comment]
Thanks @hvanhovell @amaliujia

**Pull request closed without merging**